src/htm/algorithms/Connections.cpp

```diff
@@ -75,7 +77,27 @@ void Connections::unsubscribe(UInt32 token) {
   eventHandlers_.erase(token);
 }
 
-Segment Connections::createSegment(CellIdx cell) {
+Segment Connections::createSegment(const CellIdx cell, 
+	                           const SegmentIdx maxSegmentsPerCell,
+				   const UInt32 iteration) { //TODO move iteration to Connections.iteration_ ?
```

Comment:
Can `maxSegmentsPerCell` be given to the constructor?  That way the caller only needs to pass this to connections once, instead of every time they want to make a segment.

Reply:
Can you also do the TODO note: `//TODO move iteration to Connections.iteration_` because it will simplify the connections API, and because it should be simple to do?

Reply:
> Can maxSegmentsPerCell be given to the constructor? 

I was thinking about this. This allows differents cells (w diff num segments), if that makes sense? 
And one can change number of segments on cell during runtime, ie pruning some segments on one cell. Or Allowing to add more at some stage.

Reply:
> Can you also do the TODO note: //TODO move iteration to Connections.iteration_  

actually I got stuck on that one. I would like that functionality, but Connections don't have a single unified `compute`, rather `adaptSegment` for SP, and several `computeActivity(xxx)` for TM. 
...ok, would the minimal version of `computeActivity` be a good adept for the iteration++?